### PR TITLE
Fix servicemonitor

### LINKED ISSUE #247: Service monitor doesn't match the Atlantis pods
The labels selector in the `servicemonitor` resource doesn't match the labels of the Atlantis pods, no metrics are collected, and the   `up` metric fires the TargetDown alarm.

```
➜ k get servicemonitor -n atlantis atlantis -o jsonpath='{.spec.selector.matchLabels}'
{"app":"atlantis","chart":"atlantis-4.7.0","heritage":"Helm","release":"atlantis"}
➜ k get pods -n atlantis atlantis-0 -o jsonpath='{.metadata.labels}'
{"app":"atlantis","controller-revision-hash":"atlantis-598dd97b58","release":"atlantis","statefulset.kubernetes.io/pod-name":"atlantis-0"}
```

```
➜ k exec -n monitoring prometheus-kube-prometheus-stack-prometheus-0 -- promtool query series --match='{job="atlantis"}' http://localhost:9090 2>&1 | sed 's/.*__name__="\([^"]*\)", .*/\1/'
ALERTS
ALERTS
ALERTS_FOR_STATE
count:up0
scrape_duration_seconds
scrape_samples_post_metric_relabeling
scrape_samples_scraped
scrape_series_added
up
```

The labels selector for the servicemonitor should be updated.

### PR DESCRIPTION
**Pull request by ndegory**
## what

- Fix the servicemonitor when basic auth is enabled
- Add clarification that prometheus metrics have to be enabled in the repoConfig
- Use the chart appVersion as default tag
- Build cleanup
- Update documentation
- Bump chart patch version

## why

- metrics were missing, and TargetDown was triggered for the Atlantis service

## tests

- [x] I have tested my changes by modifying the servicemonitor from a previous version of the Atlantis chart
- [x] checked the servicemonitor rendered by this PR by running helm template, with different options set in `servicemonitor.auth`
- [x] checked the image tag change:
  
  ```
  ➜ helm template atlantis ./charts/atlantis | ag image:
            image: "ghcr.io/runatlantis/atlantis:v0.22.3"
        image: dduportal/bats:0.4.0
        image: lachlanevenson/k8s-kubectl:v1.4.8-bash
  ```

## references

- originally for #247

### CODE REVIEW

File: charts/atlantis/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
-appVersion: v0.21.0
+appVersion: v0.22.0
```
Comment (nitrocode) [inline charts/atlantis/Chart.yaml:2]: The appversion is not the latest version here

File: charts/atlantis/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 description: A Helm chart for Atlantis https://www.runatlantis.io
 name: atlantis
-version: 4.9.2
+version: 4.9.3
 keywords:
```
Comment (nitrocode) [inline charts/atlantis/Chart.yaml:6]: The app version should bump the minor version to keep it consistent with previous helm chart releases. We should do this in a separate pr

File: charts/atlantis/Chart.yaml (hunk 1 above)
Comment (ndegory) [reply charts/atlantis/Chart.yaml:2]: appVersion in chart v4.9.2 was not reflecting the app version deployed by default, because the values.yaml was setting the tag to v0.22.0, I just wanted to realign these two values to avoid any confusion. I'd prefer not to update to a new version here, as I haven't tested it.

File: charts/atlantis/Chart.yaml (hunk 2 above)
Comment (ndegory) [reply charts/atlantis/Chart.yaml:6]: sure, will remove it

Comment (nitrocode) [reply charts/atlantis/Chart.yaml]: Let's just bump the minor version since the changes youre doing are still present, no?

Ref https://github.com/runatlantis/helm-charts/pull/221/files

Comment (nitrocode) [reply charts/atlantis/Chart.yaml]: I'm not sure if this will break existing functionality so I will wait for others to review this PR

File: charts/atlantis/values.yaml [hunk 3]
```diff
@@ -430,6 +431,7 @@ containerSecurityContext: {}
 
 servicemonitor:
   enabled: false
+  interval: "30s"
```
Comment (Teko012) [inline charts/atlantis/values.yaml:434]: Could you add this to the readme as well?

File: charts/atlantis/Chart.yaml (hunk 2 above)
Comment (ndegory) [reply charts/atlantis/Chart.yaml:6]: I updated the patch version, not sure if the changes in the values.yaml file require an update of the minor version.

**Comment by ndegory** [comment]
@nitrocode the PR is ready for review

File: charts/atlantis/values.yaml (hunk 3 above)
Comment (ndegory) [reply charts/atlantis/values.yaml]: done

**Comment by GenPage** [comment]
@ndegory Please rebase in the meantime, I approved the tests and linters to run.

**Comment by ndegory** [comment]
@GenPage I rebased the PR and bumped the chart version.

**Review by GenPage — APPROVED** [review]
Thank you, we appreciate the additional tweaks to CI and defaulting the image tag.
